### ISSUE TITLE
loadable-components and webpack 5, incorrect content hashes in the stats file

### ISSUE THREAD

**Issue by btpoe-kr** [jaredpalmer/razzle#1587]
### Current Behavior

1. `yarn create razzle-app --example with-loadable-components my-project`
2. `cd my-project`
3. `yarn add --dev webpack@5.24.0 html-webpack-plugin@5.2.0`
4. `yarn build`

The `build/loadable-stats.json` content hashes do not align with the corresponding files' content hashes.

### Expected behavior

The `loadable-stats.json` content hashes align with the corresponding files' content hashes.

### Reproducible example

<img width="707" alt="contenthash-mismatch" src="https://user-images.githubusercontent.com/76752439/112543267-e5c81e00-8d8b-11eb-922e-229ead43d840.png">

### Suggested solution(s)

I'm assuming the issue involves the webpack config generated by razzle, but I am unsure.

### Additional context

To confirm that it was not an issue with loadable-components and webpack 5, I created a minimal `webpack.config.js` in the project and was able to successfully compile the chunks with proper content hashes.

The example repo works properly with webpack 4.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Razzle           | 4.0.3 |
| Razzle Plugins   | N/A |
| Node             | v14.15.0 |
| Browser          | N/A |
| Yarn         | 1.22.10 |
| Operating System | macOS 10.15.7 |
| TypeScript       | N/A |
| React            | 17.0.1 |
| Webpack | 5.24.0 |

**Comment by fivethreeo** [comment]
Can you post that webpack config? Try setting options.debug.config to true and compare the configs for webpack 4 and 5.

**Comment by btpoe-kr** [comment]
Here is the `webpack.config.js` file that works:

```
const LoadableWebpackPlugin = require("@loadable/webpack-plugin");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const path = require("path");

module.exports = {
  mode: "development",
  entry: "./src/client.js",
  output: {
    filename: "client.[contenthash:8].js",
    path: path.resolve(__dirname, "build/public/static/js"),
  },
  module: {
    rules: [
      {
        test: /\.jsx?$/,
        exclude: /node_modules/,
        loader: "babel-loader",
        options: {
          presets: ["razzle/babel"],
          plugins: ["@loadable/babel-plugin"],
        },
      },
      {
        test: /\.css$/,
        use: [
          {
            loader: "css-loader",
            options: {
              sourceMap: true,
              importLoaders: 1,
              modules: {
                auto: true,
                localIdentName: "[name]__[local]___[hash:base64:5]",
              },
            },
          },
        ],
      },
    ],
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: "static/css/[name].[contenthash:8].css",
      ignoreOrder: false,
      chunkFilename: "static/css/[name].[contenthash:8].chunk.css",
    }),
    new LoadableWebpackPlugin({
      outputAsset: false,
      writeToDisk: { filename: path.resolve(__dirname, "build") },
    }),
  ],
};
```

**Comment by btpoe-kr** [comment]
I ran it with the debug option enabled and these are the differences between the 2:

webpack 4:

```
resolve: {
  plugins: [
    {
      apply: [Function: nothing],
      makePlugin: [Function (anonymous)],
      moduleLoader: [Function (anonymous)],
      topLevelLoader: { apply: [Function: nothing] },
      bind: [Function (anonymous)],
      tsLoaderOptions: [Function (anonymous)],
      forkTsCheckerOptions: [Function (anonymous)],
    },
  ],
},
resolveLoader: {
  plugins: [{ apply: [Function: nothing] }],
},
output: {
  // no library property, but otherwise equal to v5
},
optimization: {
  moduleIds: "hashed",
},
noEmitOnErrors: true,
```

webpack 5:

```
resolve: {
  plugins: [],
},
resolveLoader: 
  plugins: [],
},
output: {
  library: { type: "var", name: "client" },
},
optimization: {
  moduleIds: "deterministic",
},
emitOnErrors: false,
```

**Comment by fivethreeo** [comment]
Try setting razzleOptions.debug.options to true and compare those too. This is probably a issue 🙁But why..

**Comment by sumanbh** [comment]
This is a bug in loadable-components's webpack plugin which is fixed in their main branch but still unreleased. See https://github.com/gregberge/loadable-components/issues/712

**Comment by fivethreeo** [comment]
Will update the example when the fix is released

**Comment by btpoe-kr** [comment]
Oh awesome! Thanks @sumanbh!

**Comment by ceopaludetto** [comment]
They've released v5.15.0 which includes this fix, no code changes are needed, only updates in @loadable packages

**Comment by fivethreeo** [comment]
Fixed
